### PR TITLE
12 accept a payment

### LINKED ISSUE #12: Accept a payment
**Trigger:**

Clicked the Accept Button

**Page:**

Image

**Requirements:**

- [ ] The endpoint should be '/payment/accept'
- [ ] Should be able to accept multiple payments
- [ ] The payment status should be updated in the database

### PR DESCRIPTION
**Pull request by isabellelbgn**
- accepts pending payments
- accepts multiple pending payments
- updates pending payment status to accepted in the database

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "@nestjs/platform-express": "^9.0.0",
     "@prisma/client": "^5.7.0",
     "@supabase/supabase-js": "^2.39.1",
+    "axios": "^1.6.5",
```
Comment (viverbungag) [inline package.json]: No need na yata axios, pa uninstall nlang ako hehe

File: src/payment/payment.controller.ts [hunk 2]
```diff
@@ -14,6 +16,21 @@ import { Student } from '@prisma/client';
 export class PaymentController {
   constructor(private readonly paymentService: PaymentService) {}
 
+  @Post('accept')
+  async acceptPayments(): Promise<Student[]> {
```
Comment (viverbungag) [inline src/payment/payment.controller.ts]: Dapat meron ka yatang request body na naga contain ng list of paymentIds tapos iconvert mo yon lahat from' pending' status to 'accept' status. Para syang update na functionality

**Comment by isabellelbgn** [comment]
- updated 'acceptPayments' method to accept array of paymentIds
- modified paymentcontroller to accept dto containing paymentIds in request body
- created a new dto to handle request

**Comment by rhaeyx** [comment]
thank u isabelle!

@viverbungag hi ya, pwede pa-note for the frontend na list of studentId ang ipass na list diri na endpoint kay di sure na always same ang values sa studentId ug paymentId tas studentId ang gamit na value pang query diri

**Pull request merged**